### DISCUSSION TITLE
Running a plugin with @swc/wasm-web

### DISCUSSION THREAD

**Discussion by joseBasso** [swc-project/swc#3713] [Q&A] [answered]
Hello 👋 

I'm working on a bundler that works inside a browser and I'm trying @swc/wasm-web to do so.
Anyway, my goal is to **Get a list of each module's requires** (for example, fileA ->  ['requires fileB', requires fileC]);

My first approach was to use a plugin - [https://swc.rs/docs/usage/plugins](https://swc.rs/docs/usage/plugins) - that kept track of the requires (and changed them a bit but that's not as important to me.) It worked with @swc/core `transformSync` but sadly it did not work with @swc/wasm-web, the callback added to the "plugin" option is never called. Is this supposed to work?

I'm also open to other ideas ofc.
Thank you and congratulations to the contributors! Awesome project

**Comment by kwonoj** [comment, marked as answer]
To answer the direct question: no, currently plugin support in @wasm binary is a big open question. Relative discussion is in here https://github.com/swc-project/swc/discussions/3540 .

**Comment by kwonoj** [comment]
Also this sounds like dupe of https://github.com/swc-project/swc/issues/888 if I read correctly?

**Reply by joseBasso** [reply]
Yes, very similar. I was trying to build my own solution with a plugin but a built-in dependencies analyzer would do the same thing. Thank you for the insanely fast response!
